sdk-tests/csdk-anchor-test/src/instructions/compress_accounts_idempotent.rs

```diff
@@ -0,0 +1,137 @@
+use anchor_lang::prelude::*;
+use light_sdk::{
+    compressible::{compress_account::prepare_account_for_compression, CompressibleConfig},
+    cpi::{
+        v2::{CpiAccounts, LightSystemProgramCpi},
+        InvokeLightSystemProgram, LightCpiInstruction,
+    },
+    instruction::{account_meta::CompressedAccountMetaNoLamportsNoAddress, ValidityProof},
+    LightDiscriminator,
+};
+
+/// Auto-generated by compressible_instructions macro.
+use crate::{errors::ErrorCode, instruction_accounts::*, state::*, LIGHT_CPI_SIGNER};
+
+pub fn compress_accounts_idempotent<'info>(
+    ctx: Context<'_, '_, 'info, 'info, CompressAccountsIdempotent<'info>>,
+    proof: ValidityProof,
+    compressed_accounts: Vec<CompressedAccountMetaNoLamportsNoAddress>,
+    signer_seeds: Vec<Vec<Vec<u8>>>,
+    system_accounts_offset: u8,
+) -> Result<()> {
+    let compression_config = CompressibleConfig::load_checked(&ctx.accounts.config, &crate::ID)?;
+    if ctx.accounts.rent_sponsor.key() != compression_config.rent_sponsor {
+        msg!(
+            "rent recipient passed: {:?}",
+            ctx.accounts.rent_sponsor.key()
+        );
+        msg!(
+            "rent recipient config: {:?}",
+            compression_config.rent_sponsor
+        );
+        return Err(ProgramError::from(ErrorCode::RentRecipientMismatch).into());
+    }
+
+    let cpi_accounts = CpiAccounts::new(
+        ctx.accounts.fee_payer.as_ref(),
+        &ctx.remaining_accounts[system_accounts_offset as usize..],
+        LIGHT_CPI_SIGNER,
+    );
+
+    let pda_accounts_start = ctx.remaining_accounts.len() - signer_seeds.len();
+    let solana_accounts = &ctx.remaining_accounts[pda_accounts_start..];
+
+    let mut compressed_pda_infos = Vec::new();
+    let mut pda_indices_to_close: Vec<usize> = Vec::new();
+
+    for (i, account_info) in solana_accounts.iter().enumerate() {
+        if account_info.data_is_empty() {
+            msg!("No data. Account already compressed or uninitialized. Skipping.");
+            continue;
+        }
+        if account_info.owner == &crate::ID {
+            let data = account_info.try_borrow_data()?;
+            let discriminator = &data[0..8];
+            let meta = compressed_accounts[i];
+
+            // TODO: consider CHECKING seeds.
+            match discriminator {
+                d if d == UserRecord::LIGHT_DISCRIMINATOR => {
```

Comment:
```suggestion
                UserRecord::LIGHT_DISCRIMINATOR => {
```
this should work as well